File: Solutions/347. Top K Frequent Elements/347. Top K Frequent Elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<vector<int>> buckets(nums.size()+1);
        unordered_map<int,int> um;
        for(auto& n:nums){
            ++um[n];
        }
        for(auto& [k,v]:um){
            buckets[v].push_back(k);
        }
        vector<int> res;
        for(int i=nums.size();i>=0 and res.size()<k;--i){
            if(buckets[i].size()){
                res.insert(res.end(),buckets[i].begin(),buckets[i].end());
            }
        }
        return res;
    }
};

File: Solutions/347. Top K Frequent Elements/347. Top K Frequent Elements.go
func topKFrequent(nums []int, k int) []int {
    um:=make(map[int]int)
    for _,n:=range nums{
        um[n]+=1
    }
    buckets:=make([][]int,len(nums)+1)
    for k,v:=range um{
        buckets[v]=append(buckets[v],k)
    }
    var res []int
    for i:=len(nums);i>=0 && len(res)<k;i-=1{
        if len(buckets[i])>0{
            for _,c:=range buckets[i]{
                res=append(res,c)
            }
        }
    }
    return res
}
